### ISSUE TITLE
[TabLayout] setHeight(or setSize) for TabItem's BadgeDrawable ?

### ISSUE THREAD

**Issue by lutics** [material-components/material-components-android#797]
Hello

How can I set height(or size) for BadgeDrawable in TabLayout ?

```
        TabLayoutMediator(view.tabLayout, view.viewPager, TabLayoutMediator.TabConfigurationStrategy { tab, position ->
            tab.orCreateBadge.apply {
                backgroundColor = Color.parseColor("#db635d")
                horizontalOffset = -10
            }
            tab.text = when (position) {
                0 -> "주문 내역"
                1 -> "받은 선물"
                else -> null
            }
        }).attach()
```

expected

![Screen Shot 2019-11-28 at 12 02 05 PM](https://user-images.githubusercontent.com/7390633/69773725-f6d43f80-11d6-11ea-9edf-d19806bcd877.png)

but I've got

![Screenshot_2019-11-28-14-22-47](https://user-images.githubusercontent.com/7390633/69779238-ac5cbe00-11ea-11ea-9e27-5ce52b1e8c6b.png)

**Comment by wcshi** [comment]
Hi @lutics configuring the badge size is currently not supported because our designers want to ensure that the badge is legible when a number is displayed. Do you have plans to display a number or will this be icon only?

**Comment by lutics** [comment]
@wcshi 

No, I think it will almost be a dot which is just like above

**Comment by lutics** [comment]
As I left company, I had no connection with this issue. So close the issue

**Comment by Kxtreme** [comment]
I don't know if I came on time but I found a simple patch to make it bigger, the idea is attach it and then clear its number I tested with 1.3.0:alpha02 here is a  kotlin example:
```
    private fun addBadgeNotificationToView(
        v: View,
        parent: FrameLayout,
        verticalOffset: Int = 0,
        horizontalOffset: Int = 0): BadgeDrawable {
        val badge = BadgeDrawable.create(this);
        v.viewTreeObserver.addOnGlobalLayoutListener(object: ViewTreeObserver.OnGlobalLayoutListener {
            @Override
            override fun onGlobalLayout() {

                badge.verticalOffset = verticalOffset;
                badge.horizontalOffset = horizontalOffset;
                badge.number = 0

                BadgeUtils.attachBadgeDrawable(
                    badge,
                    v,
                   parent
                );
                badge.clearNumber()
                v.viewTreeObserver.removeOnGlobalLayoutListener(this);
            }
        })

        return badge
    }
```

**Comment by JorgeMucientes** [comment]
Hi! 
I ended up here because I'm looking for a way to make the badge bigger but apparently it's not supported or haven't found how can this be achieved. 
For accessibility reasons when the user selects to display the font big/large from device settings the number inside the badge is not displayed very nicely. It would be great to be able to scale the badge accordingly or set some kind of minimum padding. For example checkout the gmail badge when font is scaled: 

![Screenshot 2021-02-09 at 10 23 44](https://user-images.githubusercontent.com/2663464/107342762-eac55c80-6ac0-11eb-9f75-bc4a05fd21dd.png)

**Comment by wcshi** [comment]
@JorgeMucientes are you observing badge text being cut off when Font Size is set to Largest?

I am not able to reproduce the gmail badge bug in your screenshot.

**Comment by JorgeMucientes** [comment]
Hi! 
The worst case scenario it's the one I showed in the screenshot. Note that not all devices support such a large font size as Samsung devices (Samsung J7 is the device used in the screenshot). I've also tested Pixel devices for example with android stock, and the font can't be set as large as in Samsun devices. It really depends on the OEM. 
In general, what I've seen for Samsun devices is that some numbers display a slight cut off like number 7 you see in the previous screenshot. But in most cases what happens is there is 0 padding between the number and the badge border but there us no visible cut off. See some more screenshots: 

![Screenshot 2021-02-23 at 08 53 59](https://user-images.githubusercontent.com/2663464/108816169-f97a3c00-75b5-11eb-9aba-59f69537341f.png)

![Screenshot 2021-02-23 at 09 04 09](https://user-images.githubusercontent.com/2663464/108816250-20d10900-75b6-11eb-87a9-3037f721640f.png)

**Comment by wcshi** [comment]
Thank you for reporting this issue, could you please send feedback to Gmail (there should be an option in the app) as well?

**Comment by JorgeMucientes** [comment]
Yes no problem! reported to the gmail team as well linking to this github issue. 

Let me know if you need any further info!

**Comment by JorgeMucientes** [comment]
Hey @wcshi! Friendly ping. Can we close this issue now?

**Comment by OtreblaSetroc** [comment]
Hi Guys! @JorgeMucientes Do you have any updates? I'm facing a similar Issue, the number of badge is cutting off when I increase the font size.
